mesosphere/spark-build
# [COPS-3550] Replace hardcoded VIP with service.name

## What changes were proposed in this pull request?
Resolves [COPS-3550](https://jira.mesosphere.com/browse/COPS-3550) && [DCOS-39751](https://jira.mesosphere.com/browse/DCOS-39751)

Prior to this fix, VIP endpoints were hardcoded to `spark-dispatcher` and if you had multiple spark dispatchers running you would not be able to reach any specific one.

## How was this patch tested?
- Added an integration test for verifying that unique VIPs are reachable

### To test manually:
- upload a stub to your test cluster (you can the latest stub created from this PR's TC job)
- install one or more spark dispatchers
- verify that the VIPs in the endpoints tab are unique (and not set to `spark-dispatcher`)

see screenshots:

### Before
![screen shot 2018-09-21 at 1 58 57 pm](https://user-images.githubusercontent.com/4225547/45905806-95306e00-bda6-11e8-881d-032d7e40ab0e.png)

### After
![screen shot 2018-09-21 at 1 59 11 pm](https://user-images.githubusercontent.com/4225547/45905805-95306e00-bda6-11e8-94b4-13959690deb1.png)
